geode-core/src/main/java/org/apache/geode/cache/configuration/RegionConfig.java

```diff
@@ -875,161 +832,17 @@ public String getType() {
      *
      */
     public void setType(String value) {
-      this.type = value;
-    }
-
-
-    /**
-     * <p>
-     * Java class for anonymous complex type.
-     *
-     * <p>
-     * The following schema fragment specifies the expected content contained within this class.
-     *
-     * <pre>
-     * &lt;complexType>
-     *   &lt;complexContent>
-     *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType">
-     *       &lt;attribute name="expression" use="required" type="{http://www.w3.org/2001/XMLSchema}string" />
-     *       &lt;attribute name="from-clause" use="required" type="{http://www.w3.org/2001/XMLSchema}string" />
-     *       &lt;attribute name="imports" type="{http://www.w3.org/2001/XMLSchema}string" />
-     *     &lt;/restriction>
-     *   &lt;/complexContent>
-     * &lt;/complexType>
-     * </pre>
-     *
-     *
-     */
-    @XmlAccessorType(XmlAccessType.FIELD)
-    @XmlType(name = "")
-    public static class Functional {
-
-      @XmlAttribute(name = "expression", required = true)
-      protected String expression;
-      @XmlAttribute(name = "from-clause", required = true)
-      protected String fromClause;
-      @XmlAttribute(name = "imports")
-      protected String imports;
-
-      /**
-       * Gets the value of the expression property.
-       *
-       * possible object is
-       * {@link String }
-       *
-       */
-      public String getExpression() {
-        return expression;
-      }
-
-      /**
-       * Sets the value of the expression property.
-       *
-       * allowed object is
-       * {@link String }
-       *
-       */
-      public void setExpression(String value) {
-        this.expression = value;
-      }
-
-      /**
-       * Gets the value of the fromClause property.
-       *
-       * possible object is
-       * {@link String }
-       *
-       */
-      public String getFromClause() {
-        return fromClause;
-      }
-
-      /**
-       * Sets the value of the fromClause property.
-       *
-       * allowed object is
-       * {@link String }
-       *
-       */
-      public void setFromClause(String value) {
-        this.fromClause = value;
-      }
-
-      /**
-       * Gets the value of the imports property.
-       *
-       * possible object is
-       * {@link String }
-       *
-       */
-      public String getImports() {
-        return imports;
-      }
-
-      /**
-       * Sets the value of the imports property.
-       *
-       * allowed object is
-       * {@link String }
-       *
-       */
-      public void setImports(String value) {
-        this.imports = value;
+      if ("range".equalsIgnoreCase(value) || "hash".equalsIgnoreCase(value)) {
```

Comment:
Unless it is going to refer to an `enum`, I don't think validation is the responsibility of the configuration container.  Leave it to the `XSD` verification during marshaling.

Reply:
I'm a little worried about the responsibility of these classes creeping outward, but I can see your point.

Looking at `o.a.g.cache.query.IndexType`, it looks like this will need to support `"key"` also?

Reply:
The entire IndexType is deprecated. user can create either a "key" or "range" index, which will be reflected by the boolean value of Index.keyIndex,  if that value is false, it should by default be "range", but if user specified a "hash" type (deprecated), our current code still support creating it.